tests/python/23_bandwidth/host_mem_23_bandwidth.py

```diff
@@ -17,80 +17,57 @@
+import pyxrt
 
 # utils_binding.py
 sys.path.append('../')
 from utils_binding import *
 
-# Define libc helpers
-libc_name = ctypes.util.find_library("c")
-libc = ctypes.CDLL(libc_name)
-libc.memcmp.argtypes = (ctypes.c_void_p, ctypes.c_void_p, ctypes.c_size_t)
-libc.memcmp.restype = (ctypes.c_int)
-libc.memcpy.argtypes = (ctypes.c_void_p, ctypes.c_void_p, ctypes.c_size_t)
-libc.memcpy.restype = (ctypes.c_void_p)
-
 current_micro_time = lambda: int(round(time.time() * 1000000))
 
-globalbuffersize = 1024*1024*2    #2 MB
+DATASIZE = int(1024*1024*2)    #2 MB
 
 def getThreshold(devHandle):
     threshold = 30000
-    deviceInfo = xclDeviceInfo2()
-    xclGetDeviceInfo2(devHandle, ctypes.byref(deviceInfo))
-    if b"qdma" in deviceInfo.mName or b"qep" in deviceInfo.mName:
+    name = devHandle.get_info(pyxrt.xrt_info_device.name)
+
+    if "qdma" in name or "qep" in name:
         threshold = 30000
-    if b"gen3x4" in deviceInfo.mName or b"_u26z_" in deviceInfo.mName:
+    if "gen3x4" in name or "_u26z_" in name:
         threshold = 20000
-    if b"u2x4" in deviceInfo.mName or b"U2x4" in deviceInfo.mName or b"u2_gen3x4" in deviceInfo.mName:
+    if "u2x4" in name or "U2x4" in name or "u2_gen3x4" in name:
         threshold = 10000
-    if b"_u25_" in deviceInfo.mName or b"_u30_" in deviceInfo.mName: # so that it doesn't set theshold for u250
+    if "_u25_" in name or "_u30_" in name: # so that it doesn't set theshold for u250
         threshold = 9000
     return threshold
 
 def getInputOutputBuffer(devhdl, krnlhdl, argno, isInput):
-    grpid = xrtKernelArgGroupId(krnlhdl, argno)
-    if grpid < 0:
-        raise RuntimeError("failed to find BO group ID: %d" % grpid)
-    # To alloc HOST only buffer, we have to specify flags XCL_BO_FLAGS_HOST_ONLY = 1<<29
-    bo = xrtBOAlloc(devhdl, globalbuffersize, 1<<29, grpid)
-    if bo == 0:
-        raise RuntimeError("failed to alloc buffer")
-
-    bobuf = xrtBOMap(bo)
-    if bobuf == 0:
-        raise RuntimeError("failed to map buffer")
-    lst = ctypes.cast(bobuf, ctypes.POINTER(ctypes.c_char))
+    bo = pyxrt.bo(devhdl, DATASIZE, pyxrt.bo.normal, krnlhdl.group_id(argno))
+    buf = bo.map()
 
-    for i in range(globalbuffersize):
-        lst[i] = i%256 if isInput else 0
+    for i in range(DATASIZE):
+        buf[i] = i%256 if isInput else 0
 
-    xrtBOSync(bo, xclBOSyncDirection.XCL_BO_SYNC_BO_TO_DEVICE, globalbuffersize, 0)
-    return bo, bobuf
+    bo.sync(pyxrt.xclBOSyncDirection.XCL_BO_SYNC_BO_TO_DEVICE, DATASIZE, 0)
+    return bo, buf
 
 def runKernel(opt):
-    try:
-        khandle3 = xrtPLKernelOpen(opt.handle, opt.xuuid, "bandwidth3")
-    except Exception as e:
-        return errno.EOPNOTSUPP
+    d = pyxrt.device(opt.index)
+    xbin = pyxrt.xclbin(opt.bitstreamFile)
+    uuid = d.load_xclbin(xbin)
 
-    kfunc = xrtKernelGetFunc(xrtBufferHandle, xrtBufferHandle, ctypes.c_int, ctypes.c_int)
+    khandle3 = pyxrt.kernel(d, uuid, "bandwidth3", pyxrt.kernel.shared)
```

Comment:
This was under an exception earlier. Please keep under exception. I dont think all platforms contains bandwidth3 kernel

Reply:
Thank you, I've moved it back under the exception.